CountCoinChange.java

```diff
@@ -12,21 +12,23 @@ public int change(int amount, int[] coins) {
             dp[0][j] = 9999;
 
         //calc dp array
-        for(int i=1;i<dp.length;i++){
-            for(int j=1;j<dp[0].length;j++){
-
-                //copy from top
-                if(j < coins[i-1]){
-                    dp[i][j] = dp[i-1][j];
-                }
-                //compare and calc
-                else{
-                    dp[i][j] = Math.min(dp[i-1][j], dp[i][j - coins[i-1]] + 1);
-                }
-
-                System.out.print(dp[i][j] + " ");
-            }
-            System.out.println();
+        // for(int i=1;i<dp.length;i++){
+        //     for(int j=1;j<dp[0].length;j++){
+
+        //         //copy from top
+        //         if(j < coins[i-1]){
+        //             dp[i][j] = dp[i-1][j];
+        //         }
+        //         //compare and calc
+        //         else{
```

Comment:
You are doing something different here . Right ?

Reply:
Yes, Ashmeet. I started out thinking I should do some steps after filling the matrix (like a traversal to determine number of combinations). Then realized the formula to fill dp matrix is different, so updated it in the final commit.